### ISSUE TITLE
[BUG]: Cast List Not Populating Correctly from IAFD Match

### ISSUE THREAD

**Issue by jocklover** [CodyBerenson/PGMA-Modernized#171]
Seeing an issue where an IAFD match is not returning correct cast info. Seeing "catch phrases" instead of cast names.

File Name: (Active Duty) - Below Deck (2007) - [Adam Wells, Kaden Saylor, Levi]
**NOTE:** Due to this failure, I attempted manual cast listing, which also failed, opening separate bug for this.
URL: https://www.iafd.com/title.rme/title=below+deck/year=2007/below-deck.htm
(Also occurred with Below Deck 3: https://www.iafd.com/title.rme/title=below+deck+3/year=2007/below-deck-3.htm)

Plex Screenshot:
![image](https://user-images.githubusercontent.com/57208899/170638204-05be5c41-007b-4a40-aeb1-cb0171f61f84.png)

IAFD Agent Log:
[Incorrect Cast List from IAFD.txt](https://github.com/CodyBerenson/PGMA-Modernized/files/8784260/Incorrect.Cast.List.from.IAFD.txt)
As noted above, match was ran w/ casting override, but Log still shows Cast List from IAFD pulled down, but apparently doesn't make it into Plex metadata.

THANK YOU

**Comment by JPH71** [comment]
Which agent are you using - this can not be IAFD, is it fagalicious, queerclick or another scene agent?

**Comment by jocklover** [comment]
This is IAFD, see log in original post. Cheers

**Comment by JPH71** [comment]
I have yet to get to iafd...
I have done 5 so far

**Comment by jocklover** [comment]
I think I have confuddle things w/ 2 similar bugs open - this is a bug where the IAFD downloaded cast list is not populating correctly. This isn't related to the manual cast list in title.

**Comment by JPH71** [comment]
OK.... point me to the correct issue and when I get to IAFD, I will make
sure that its sorted...
I can't remember if you gave me an example where the issue happened... if
not please do. It makes life much more easier when I have an example and a
log file...

**Comment by jocklover** [comment]
This issue is where the Cast list shown in Plex is not the cast list downloaded from IAFD. Log shows it grabbed the cast list, Plex just shows a few "catchphrases" instead. Examples & log are above.
Thanks for your work on this!

**Comment by JPH71** [comment]
Just looked at this more in depth.
You were right it was IAFD that scraped it last, but the cast list did not come from IAFD..
When you look at the Cast - "Colin taking Levi...", if you look through the log file this text can not be found.
I suspect you must have scraped against another agent, then changed the file name. 
When you do this Plex accepts the file name change on disk but does not recreate a new ID, so it carries over previously scraped data over.
To get a clean scrape you will have to take the film file out of the directory and scan the library to remove the old metadata then put the file back in and let plex scrape afresh.

However, incidentally the other day when I was working on GEVI, I noticed this issue and I have jiggled the code so that it clears out directors and cast then scrapes... this means we are assured that every time we run the agent we are having fresh results.
I have therefore made the necessary changes to IAFD.

Thanks for picking up this error... In the meantime do the Plex Dance with this title and check the results
